compiler/src/iree/compiler/Codegen/SPIRV/VideoCoreConfig.cpp

```diff
@@ -28,6 +28,119 @@ using CodeGenPipeline =
     mlir::iree_compiler::IREE::Codegen::DispatchLoweringPassPipeline;
 namespace mlir::iree_compiler::detail {
 
+static int64_t findLargestPowerOfTwoMultiple(int64_t dim, int64_t startValue) {
+  if (dim < 0) {
+    return 1;
+  }
+  while (dim % startValue != 0) {
+    startValue >>= 1;
+  }
+  return std::max(startValue, 1ll);
+}
+
+LogicalResult
+setMatmulOpVideoCoreConfig(IREE::GPU::TargetAttr target, linalg::LinalgOp op,
```

Comment:
Can you add as many comments as possible to this function? Again try to think of someone who never optimised the tiling

Reply:
I've added as much as I could. I actually want to revisit this on one of our Fridays to improve it a bit. I think I have some ideas on how to find a better tiling.